galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/commons/ExtractQuerySort.java

```diff
@@ -3,60 +3,38 @@
  */
 package dev.galasa.framework.api.ras.internal.commons;
 
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import javax.servlet.http.HttpServletResponse;
+
+import static dev.galasa.framework.api.ras.internal.commons.ServletErrorMessage.*;
 
 public class ExtractQuerySort {
 	
 	/**
-	 * Given a map of parameters from the HTTP query, look up all the 
-	 * ?sort=xxx and ?sort=xxx:asc ?sort=yyy:desc ?sort=xxx:asc,yyy:desc
-	 * 
-	 * Create a map which contains simplified information of:
-	 * { "xxx" : TRUE, "yyy" : FALSE }
+	 * Check if the sort value contains an asc for ascending or desc for descending
+	 * otherwise return null
 	 */
-	public static Map<String, Boolean> extractParameters(Map<String, String[]> query){
 		
-		Map<String, Boolean> paramMap = new HashMap<>();
-    	//checks to see if sort is valid
-    	if(query.containsKey("sort")) {
-    		//retrieves parameters
-    		String[] params = query.get("sort");
-    		if(params != null) {
-    			//for the value in the parameters, split at ','
-	    		for(String value : params) {
-	    			List<String> val = Arrays.asList(value.split(","));
-	    			if(val != null) {
-	    				for (String item : val) {
-		    				if(item.contains(":")) {
-		    					List<String> split = Arrays.asList(item.split(":"));
-		    					if(split.get(1).equals("desc")) {
-		    						paramMap.put(split.get(0), Boolean.FALSE);
-		    					}else {
-		    						paramMap.put(split.get(0), Boolean.TRUE);
-		    					}
-	    					}
-	    				}
-	    			}
-	    		}
-    		}
-    	}
-    	
-    	return paramMap;
-    }
-	
-	public static Boolean isAscending(QueryParameters queryParams, String param) {
-		try {
-			if (queryParams.getSingleString("sort", null) != null) {
-				String extractedParam = queryParams.getSingleString(param, null);
-				return extractedParam != null;
+	public static Boolean isAscending(QueryParameters queryParams, String param) throws InternalServletException{
+		if (queryParams.getSingleString("sort", null) != null) {
+			String sortValue = queryParams.getSingleString("sort", null);
```

Comment:
Sounds like this is just getting multiple values from the queryParams, with a delimeter of ":" ? If we did that most of this logic would fit nicely into the QueryParameter class, as it extracts values from query parameters already.

Reply:
And we'd avoid a static method. I'd rather it was an object with a method rather than static also.

Reply:
I believe there is an issue to look at this in more detail than what has been done here, as this is only to allow the age query to work